generator/js_generator.cc

```diff
@@ -3707,18 +3709,20 @@ void Generator::GenerateExtension(const GeneratorOptions& options,
 
   printer->Print(
       "\n"
+      "if(!$extendName$Binary) {\n"
```

Comment:
@gonzojive, @egormodin: this "if not, then set" block (as well as the one below on line 3740) is questionable to me... the previous method was able to assign directly into `proto...` without first ensuring an array existed there.

Do you think there's a better or more elegant way to ensure the underlying assignment target exists as an array?

Reply:
Indeed, this was a sign something was wrong. Your subsequent changes made this unnecessary, I believe.